File: valet_parking_app/lib/models/vehicle.dart
class Vehicle {
  final String id;
  final String ownerId;
  final String registrationNumber;
  final String make;
  final String model;
  final String color;
  final DateTime createdAt;

  Vehicle({
    required this.id,
    required this.ownerId,
    required this.registrationNumber,
    required this.make,
    required this.model,
    required this.color,
    required this.createdAt,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] ?? '',
      ownerId: json['owner_id'] ?? '',
      registrationNumber: json['registration_number'] ?? '',
      make: json['make'] ?? '',
      model: json['model'] ?? '',
      color: json['color'] ?? '',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'owner_id': ownerId,
      'registration_number': registrationNumber,
      'make': make,
      'model': model,
      'color': color,
      'created_at': createdAt.toIso8601String(),
    };
  }

  String get displayName => '$make $model ($registrationNumber)';
}

File: backend/internal/models/vehicle.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type VehicleType string

const (
	VehicleTypeCar        VehicleType = "car"
	VehicleTypeBike       VehicleType = "bike"
	VehicleTypeThreeWheel VehicleType = "three_wheeler"
)

type Vehicle struct {
	ID                 primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	OwnerID            primitive.ObjectID `bson:"owner_id" json:"owner_id"`
	RegistrationNumber string             `bson:"registration_number" json:"registration_number"`
	Make               string             `bson:"make" json:"make"`
	Model              string             `bson:"model" json:"model"`
	Color              string             `bson:"color" json:"color"`
	VehicleType        VehicleType        `bson:"vehicle_type" json:"vehicle_type"`
	Photos             []string           `bson:"photos,omitempty" json:"photos,omitempty"`
	CreatedAt          time.Time          `bson:"created_at" json:"created_at"`
}
